### PR TITLE
Changing logic to create private channels instead of IM Groups

### PR DESCRIPTION
**Pull request by stefdworschak**
## Description
Creating Private Slack Channels instead of IM Groups to better manage drop-offs and other team member changes.

## Pull request type

- [x] #M01 (Miscellaneous User Story #01)
- [ ] #P02 (Participant User Story #02)
- [ ] #S03 (Staff User Story #03)
- [ ] #A04 (Admin User Story #04)
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Related Issue
#275 

## Configuration instructions
- Add `SLACK_TEAM_ID` to `.env` file

### CODE REVIEW

File: teams/views.py [hunk 1]
```diff
@@ -213,66 +219,106 @@ def rename_team(request, team_id):
...
-            or settings.SLACK_WORKSPACE):
+    if (not (settings.SLACK_ENABLED or settings.SLACK_BOT_TOKEN
+            or settings.SLACK_WORKSPACE) or slack_site_settings.communication_channel_type != 'slack_private_channel'):
         messages.error(request, 'This feature is currently not enabled.')
         return redirect(reverse('view_team', kwargs={'team_id': team_id}))
 
+    # Create new channel
+    date_str = datetime.now().strftime('%y%m')
+    team_name = re.sub('[^A-Za-z0-9]+', '', team.display_name.lower())
+    channel_name = f'{date_str}-hackathon-{team_name}'
```
Comment (TravelTimN) [inline teams/views.py:250]: brilliant! this was going to be my first question... what're the private channel names and convention... nice! 👍🏻

**Review by TravelTimN — APPROVED** [review]
LGTM! 👍🏻 
Maybe I've missed it somewhere in there, but what happens if the team name changes, and an existing private channel was already created before they changed the team name?
Clicking on the button to go to their private channel, does that then attempt to create a new private channel?

Also, how are the archives managed after a Hackathon ends?
Will these be cleaned-up and removed from Slack after `N` days/weeks/months/year?

**Comment by stefdworschak** [comment]
@TravelTimN thanks for the feedback here! So for the first coupe of questions:
- As you've seen, yes there is a naming convention based on the hackathon and the team name to make it as unique, but also understandable as possible.
- Once the channel is created, if the team name is changed the channel name (witth this implementation) will not change, but the members can always change it in Slack
- Clicking the button will create a new channel if the "communication_channel" field for the team is empty otherwise just open the link to the channel
- In  terms of archives, that's more of a Community question, but from a platform perspective the link will be there so it will try to redirect to their channel and if the channel is archived I guess give an error message in Slack.

File: teams/views.py (hunk 1 above)
Comment (stefdworschak) [reply teams/views.py:250]: Thanks :+1:

**Pull request merged**